Dataset source: https://data.mendeley.com/datasets/jwyy9np4gv/3

Recorded using anelectronic stethoscope 3200, 3M Littmann.

```
The system provides a built-in ambient and frictional reduction technology. All signals were acquired at a sampling rate of 4 kHz using a 16-bit quantizer and were bandlimited to an extended frequency range of 20-2 kHz. Moreover, the frequency response between 50 and 500 Hz was attenuated at the acquisition stage to minimize the interference of heart beats sounds.
```

It is also note-worthy that the 3M Littman electronic stethoscope is capable of noise reduction.

```
    Active Noise Cancellation
    
Help cut out distracting noises common to hectic environments with our  active noise cancellation technology. It can help you listen by reducin  unwanted outside background noise so you can focus on listening to what’s important: your patient and their body sounds. Hear the difference
```
Source: https://www.littmann.com/3M/en_US/littmann-stethoscopes/advantages/core-digital-stethoscope/

In [18]:
import numpy as np
import matplotlib.pyplot as plt
import os

In [ ]:
types = os.listdir('./data')

In [34]:
for type in types:
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        if audio_file[-3:] != 'wav':
            print(audio_file, type)

desktop.ini Asthma


`desktop.ini` found in Asthma folder.

In [35]:
for type in types:
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        if audio_file[-3:] != 'wav':
            os.remove(f'./data/{type}/{audio_file}')

In [66]:
for type in types:
    lent = len(os.listdir(f'./data/{type}'))
    print(f'{type}: {lent} samples')

Asthma: 99 samples
Normal: 105 samples
Pneumonia: 15 samples
Pulmonary Congestion: 63 samples


In [38]:
sampling_rate_dictionary = {}

for type in types:
    sampling_rate_dictionary[type] = {}
    # print(os.listdir(f'./data/{type}')[-1])
    for audio_file in os.listdir(f'./data/{type}'):
        signal, sr = librosa.load(f'./data/{type}/{audio_file}')
        if str(sr) not in sampling_rate_dictionary[type]:
            sampling_rate_dictionary[type][str(sr)] = 1
        else:
            sampling_rate_dictionary[type][str(sr)] += 1

In [39]:
for type in types:
    print(sampling_rate_dictionary[type])

{'22050': 99}
{'22050': 105}
{'22050': 15}
{'22050': 63}


As we can see, all sampling rate for all audio files are in 22050 hertz.

We proceed to check the average length of the audio files.

In [59]:
import pandas as pd

def func(x):
    signal, sr = librosa.load(x)
    return len(signal)/sr

for type in types:
    audio_length_series = pd.Series(os.listdir(f'./data/{type}'))
    prefix_series = pd.Series([f'./data/{type}/' for i in range(audio_length_series.size)])
    audio_length_series = prefix_series+audio_length_series
    audio_length_series = audio_length_series.apply(func)
    print(f'{type}')
    print(pd.Series(audio_length_series.value_counts().index.tolist()).describe())

Asthma
count    29.000000
mean     16.296847
std       5.408483
min       5.936009
25%      12.528027
50%      15.344036
75%      19.888027
max      30.000000
dtype: float64
Normal
count    34.000000
mean     17.971783
std       4.948170
min       9.840000
25%      13.600023
50%      17.904014
75%      21.584036
max      27.888027
dtype: float64
Pneumonia
count     5.000000
mean     19.580816
std       9.024375
min       9.072018
25%      11.184036
50%      23.472018
75%      24.176009
max      30.000000
dtype: float64
Pulmonary Congestion
count    20.000000
mean     17.049621
std       5.676013
min       8.368027
25%      13.072029
50%      15.248005
75%      22.160023
max      27.568027
dtype: float64


The average lung sounds for all types is between 16 and 20 seconds.

The maximum recording length is at 30 seconds and has a minimum recording length between 6 and 10 seconds.

The researchers can opt to adjust the recording length of each audio files then apply data augmentation to increase the amount/size of each type.

The original researchers that utilized the dataset proceeded to do feature extraction right after the acquisition of data.